src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -148,10 +148,10 @@ internal sealed class CachedCodeEntry
 #endif
             public int _capsize;
             public ExclusiveReference _runnerref;
-            public SharedReference _replref;
+            public WeakReference<RegexReplacement> _replref;
```

Comment:
It would be nice to make unrelated changes in separate PRs so that big changes like this are easier to review.

Reply:
You are right, I should put that change back into a separate PR where it was to begin with. I paid attention to not add unrelated work but failed with this one.

Reply:
Created https://github.com/dotnet/corefx/pull/27791 for the SharedReference work.

Reply:
Can you push up a change to eliminate this from the diff now maybe

Reply:
done